unsigned int tomo_db0::IoLoop::epoll_pass<false, true, false>(int)
--- PRE

+++ POST

@@ -2,7 +2,7 @@

 0004 [ 2] push %r15
 0006 [ 2] push %r14
 0008 [ 2] push %r13
-000a [ 3] mov %rdi,%r15
+000a [ 3] mov %rdi,%r14
 000d [ 2] push %r12
 000f [ 1] push %rbp
 0010 [ 1] push %rbx
@@ -20,197 +20,191 @@

 0044 [ 8] movl $0x0,0xc(%rsp)
 004c [ 2] test %eax,%eax
 004e [ 4] cmovns %eax,%r12d
-0052 [ 7] mov %r12d,0xc18(%r15)
-0059 [ 6] jle <+0x14f>
-005f [ 7] mov 0x190(%r15),%rax
-0066 [ 3] movslq %r12d,%rdx
-0069 [ 2] xor %ebp,%ebp
-006b [ 7] lea <rip>(%rip),%r13 # <_IO_stdin_used+0x16e90>
-0072 [ 7] add %rdx,0x520(%rax)
-0079 [ 7] lea 0xac8(%r15),%rax
-0080 [ 5] mov %rax,0x10(%rsp)
-0085 [11] data16 cs nopw 0x0(%rax,%rax,1)
-0090 [ 4] mov 0x4(%rbx),%rdx
-0094 [ 3] mov %rdx,%rax
+0052 [ 7] mov %r12d,0xc18(%r14)
+0059 [ 2] jle <+0xcb>
+005b [ 7] mov 0x190(%r14),%rax
+0062 [ 3] movslq %r12d,%rdx
+0065 [ 2] xor %ebp,%ebp
+0067 [ 7] lea <rip>(%rip),%r13 # <_IO_stdin_used+0x16fb0>
+006e [ 7] add %rdx,0x520(%rax)
+0075 [ 7] lea 0xac8(%r14),%rax
+007c [ 5] mov %rax,0x10(%rsp)
+0081 [11] data16 cs nopw 0x0(%rax,%rax,1)
+008c [ 4] nopl 0x0(%rax)
+0090 [ 4] mov 0x4(%rbx),%rsi
+0094 [ 3] mov %rsi,%rax
 0097 [ 4] shr $0x38,%rax
 009b [ 2] cmp $0x11,%al
-009d [ 6] ja <+0x140>
-00a3 [ 5] movslq 0x0(%r13,%rax,4),%rax
-00a8 [ 3] add %r13,%rax
-00ab [ 3] notrack jmp *%rax
-00ae [ 2] mov (%rbx),%eax
-00b0 [ 5] test $0x2019,%eax
-00b5 [ 2] je <+0xbb>
-00b7 [ 4] movb $0x0,0x29(%rdx)
-00bb [ 2] test $0x1c,%al
-00bd [ 2] je <+0xec>
-00bf [ 4] cmpb $0x0,0x30(%rdx)
-00c3 [ 5] mov %rdx,0x20(%rsp)
-00c8 [ 2] jne <+0xec>
-00ca [ 5] lea 0x20(%rsp),%r14
-00cf [ 7] lea 0x198(%r15),%rdi
-00d6 [ 4] movb $0x1,0x30(%rdx)
-00da [ 5] mov %rdx,0x18(%rsp)
-00df [ 3] mov %r14,%rsi
-00e2 [ 5] call <std::deque<tomo_db0::Client*, std::allocator<tomo_db0::Client*> >::push_back(tomo_db0::Client* const&)>
-00e7 [ 5] mov 0x18(%rsp),%rdx
-00ec [ 6] mov 0x7bc(%rdx),%eax
-00f2 [ 3] cmp $0xffffffff,%eax
-00f5 [ 2] je <+0x124>
-00f7 [ 7] mov 0x1a68(%r15),%rsi
-00fe [ 7] mov 0x1a70(%r15),%rcx
-0105 [ 3] sub %rsi,%rcx
-0108 [ 4] sar $0x3,%rcx
-010c [ 3] cmp %rcx,%rax
-010f [ 2] jae <+0x124>
-0111 [ 4] mov (%rsi,%rax,8),%rax
-0115 [ 3] test %rax,%rax
-0118 [ 2] je <+0x124>
-011a [ 6] movw $0x0,0x3e(%rax)
-0120 [ 4] movb $0x0,0x29(%rdx)
-0124 [ 3] mov %rdx,%rsi
-0127 [ 3] mov %r15,%rdi
-012a [ 5] call <tomo_db0::IoLoop::mark_active(tomo_db0::Client*)>
-012f [ 1] nop
-0130 [ 4] incl 0xc(%rsp)
-0134 [11] data16 cs nopw 0x0(%rax,%rax,1)
-013f [ 1] nop
-0140 [ 2] inc %ebp
-0142 [ 4] add $0xc,%rbx
-0146 [ 3] cmp %r12d,%ebp
-0149 [ 6] jl <+0x90>
-014f [ 5] mov 0x28(%rsp),%rax
-0154 [ 9] sub %fs:0x28,%rax
-015d [ 6] jne <+0x3a8>
-0163 [ 4] mov 0xc(%rsp),%eax
-0167 [ 4] add $0x38,%rsp
-016b [ 1] pop %rbx
-016c [ 1] pop %rbp
-016d [ 2] pop %r12
-016f [ 2] pop %r13
-0171 [ 2] pop %r14
-0173 [ 2] pop %r15
-0175 [ 1] ret
-0176 [10] cs nopw 0x0(%rax,%rax,1)
-0180 [ 5] lea 0x20(%rsp),%r14
-0185 [ 9] movq $0x0,0x20(%rsp)
-018e [ 2] xchg %ax,%ax
-0190 [ 7] mov 0xbc8(%r15),%edi
-0197 [ 5] mov $0x8,%edx
-019c [ 3] mov %r14,%rsi
-019f [ 5] call <read@plt>
-01a4 [ 4] cmp $0x8,%rax
-01a8 [ 2] je <+0x190>
-01aa [ 7] mov 0x190(%r15),%rax
-01b1 [ 2] inc %ebp
-01b3 [ 4] add $0xc,%rbx
-01b7 [ 4] incl 0xc(%rsp)
-01bb [ 7] incq 0x440(%rax)
-01c2 [ 3] cmp %r12d,%ebp
-01c5 [ 6] jl <+0x90>
-01cb [ 2] jmp <+0x14f>
-01cd [ 3] nopl (%rax)
-01d0 [ 7] mov 0x190(%r15),%rax
-01d7 [ 5] mov 0x10(%rsp),%rdx
-01dc [ 7] lea <rip>(%rip),%rdi # <tomo_db0::Server::databases_>
-01e3 [ 2] inc %ebp
-01e5 [ 4] add $0xc,%rbx
-01e9 [ 2] mov (%rax),%esi
-01eb [ 5] call <tomo_db0::DatabaseMap::consume_wake(unsigned int, tomo_db0::Ring&)>
+009d [ 2] ja <+0xb0>
+009f [ 5] movslq 0x0(%r13,%rax,4),%rax
+00a4 [ 3] add %r13,%rax
+00a7 [ 3] notrack jmp *%rax
+00aa [ 6] nopw 0x0(%rax,%rax,1)
+00b0 [ 3] mov %r14,%rdi
+00b3 [ 5] call <tomo_db0::IoLoop::fliplisten_epoll(unsigned long)>
+00b8 [ 4] add %eax,0xc(%rsp)
+00bc [ 4] nopl 0x0(%rax)
+00c0 [ 2] inc %ebp
+00c2 [ 4] add $0xc,%rbx
+00c6 [ 3] cmp %r12d,%ebp
+00c9 [ 2] jl <+0x90>
+00cb [ 5] mov 0x28(%rsp),%rax
+00d0 [ 9] sub %fs:0x28,%rax
+00d9 [ 6] jne <+0x388>
+00df [ 4] mov 0xc(%rsp),%eax
+00e3 [ 4] add $0x38,%rsp
+00e7 [ 1] pop %rbx
+00e8 [ 1] pop %rbp
+00e9 [ 2] pop %r12
+00eb [ 2] pop %r13
+00ed [ 2] pop %r14
+00ef [ 2] pop %r15
+00f1 [ 1] ret
+00f2 [11] data16 cs nopw 0x0(%rax,%rax,1)
+00fd [ 3] nopl (%rax)
+0100 [10] movabs $0xffffffffffff,%rax
+010a [ 3] and %rsi,%rax
+010d [ 2] je <+0xc0>
+010f [ 6] mov 0x788(%rax),%edx
+0115 [ 7] mov 0x190(%r14),%rcx
+011c [ 2] cmp %edx,(%rcx)
+011e [ 2] jne <+0xc0>
+0120 [ 4] cmpb $0x0,0x33(%rax)
+0124 [ 2] jne <+0xc0>
+0126 [ 6] mov 0x784(%rax),%edx
+012c [ 3] cmp $0xfffffffe,%edx
+012f [ 2] je <+0xc0>
+0131 [ 7] mov 0x288(%r14),%rdx
+0138 [ 7] mov 0x290(%r14),%rcx
+013f [ 3] cmp %rcx,%rdx
+0142 [ 2] je <+0x162>
+0144 [11] data16 cs nopw 0x0(%rax,%rax,1)
+014f [ 1] nop
+0150 [ 3] cmp (%rdx),%rax
+0153 [ 6] je <+0xc0>
+0159 [ 4] add $0x28,%rdx
+015d [ 3] cmp %rdx,%rcx
+0160 [ 2] jne <+0x150>
+0162 [ 2] mov (%rbx),%edx
+0164 [ 6] test $0x2019,%edx
+016a [ 2] je <+0x170>
+016c [ 4] movb $0x0,0x29(%rax)
+0170 [ 3] and $0x1c,%edx
+0173 [ 2] je <+0x1a2>
+0175 [ 4] cmpb $0x0,0x30(%rax)
+0179 [ 5] mov %rax,0x20(%rsp)
+017e [ 2] jne <+0x1a2>
+0180 [ 5] lea 0x20(%rsp),%r15
+0185 [ 7] lea 0x198(%r14),%rdi
+018c [ 4] movb $0x1,0x30(%rax)
+0190 [ 5] mov %rax,0x18(%rsp)
+0195 [ 3] mov %r15,%rsi
+0198 [ 5] call <std::deque<tomo_db0::Client*, std::allocator<tomo_db0::Client*> >::push_back(tomo_db0::Client* const&)>
+019d [ 5] mov 0x18(%rsp),%rax
+01a2 [ 6] mov 0x7bc(%rax),%edx
+01a8 [ 3] cmp $0xffffffff,%edx
+01ab [ 2] je <+0x1da>
+01ad [ 7] mov 0x1a68(%r14),%rsi
+01b4 [ 7] mov 0x1a70(%r14),%rcx
+01bb [ 3] sub %rsi,%rcx
+01be [ 4] sar $0x3,%rcx
+01c2 [ 3] cmp %rcx,%rdx
+01c5 [ 2] jae <+0x1da>
+01c7 [ 4] mov (%rsi,%rdx,8),%rdx
+01cb [ 3] test %rdx,%rdx
+01ce [ 2] je <+0x1da>
+01d0 [ 6] movw $0x0,0x3e(%rdx)
+01d6 [ 4] movb $0x0,0x29(%rax)
+01da [ 3] mov %rax,%rsi
+01dd [ 3] mov %r14,%rdi
+01e0 [ 5] call <tomo_db0::IoLoop::mark_active(tomo_db0::Client*)>
+01e5 [11] data16 cs nopw 0x0(%rax,%rax,1)
 01f0 [ 4] incl 0xc(%rsp)
-01f4 [ 3] cmp %r12d,%ebp
-01f7 [ 6] jl <+0x90>
-01fd [ 5] jmp <+0x14f>
-0202 [11] data16 cs nopw 0x0(%rax,%rax,1)
-020d [ 3] nopl (%rax)
-0210 [ 7] mov 0x188(%r15),%rax
-0217 [ 7] movzbl 0x11460(%rax),%eax
-021e [ 2] test %al,%al
-0220 [ 6] jne <+0x140>
-0226 [ 7] mov 0xaa4(%r15),%r14d
-022d [ 3] test %r14d,%r14d
-0230 [ 6] js <+0x140>
-0236 [ 8] movl $0x0,0x18(%rsp)
-023e [ 2] xor %edx,%edx
-0240 [ 2] xor %esi,%esi
-0242 [ 5] mov $0x80800,%ecx
-0247 [ 3] mov %r14d,%edi
-024a [ 5] call <accept4@plt>
-024f [ 2] test %eax,%eax
-0251 [ 6] jns <+0x390>
-0257 [ 5] call <__errno_location@plt>
-025c [ 2] mov (%rax),%eax
-025e [ 3] cmp $0x4,%eax
-0261 [ 6] jne <+0x33f>
-0267 [ 2] jmp <+0x23e>
-0269 [ 7] nopl 0x0(%rax)
-0270 [10] movabs $0xffffffffffff,%rax
-027a [ 3] and %rax,%rdx
-027d [ 6] je <+0x140>
-0283 [ 6] mov 0x788(%rdx),%eax
-0289 [ 7] mov 0x190(%r15),%rcx
-0290 [ 2] cmp %eax,(%rcx)
-0292 [ 6] jne <+0x140>
-0298 [ 4] cmpb $0x0,0x33(%rdx)
-029c [ 6] jne <+0x140>
-02a2 [ 6] mov 0x784(%rdx),%eax
-02a8 [ 3] cmp $0xfffffffe,%eax
-02ab [ 6] je <+0x140>
-02b1 [ 7] mov 0x288(%r15),%rax
-02b8 [ 7] mov 0x290(%r15),%rcx
-02bf [ 3] cmp %rcx,%rax
-02c2 [ 2] jne <+0x2dd>
-02c4 [ 5] jmp <+0xae>
-02c9 [ 7] nopl 0x0(%rax)
-02d0 [ 4] add $0x28,%rax
-02d4 [ 3] cmp %rax,%rcx
-02d7 [ 6] je <+0xae>
-02dd [ 3] cmp (%rax),%rdx
-02e0 [ 2] jne <+0x2d0>
-02e2 [ 5] jmp <+0x140>
-02e7 [ 9] nopw 0x0(%rax,%rax,1)
-02f0 [ 7] mov 0x188(%r15),%rax
-02f7 [ 7] movzbl 0x11460(%rax),%eax
-02fe [ 2] test %al,%al
-0300 [ 6] jne <+0x140>
-0306 [ 7] mov 0xaa0(%r15),%r14d
-030d [ 3] test %r14d,%r14d
-0310 [ 6] js <+0x140>
-0316 [ 8] movl $0x0,0x18(%rsp)
-031e [ 2] xor %edx,%edx
-0320 [ 2] xor %esi,%esi
-0322 [ 5] mov $0x80800,%ecx
-0327 [ 3] mov %r14d,%edi
-032a [ 5] call <accept4@plt>
-032f [ 2] test %eax,%eax
-0331 [ 2] jns <+0x370>
-0333 [ 5] call <__errno_location@plt>
-0338 [ 2] mov (%rax),%eax
-033a [ 3] cmp $0x4,%eax
-033d [ 2] je <+0x31e>
-033f [ 3] cmp $0xb,%eax
-0342 [ 2] je <+0x352>
-0344 [ 7] mov 0x190(%r15),%rax
-034b [ 7] incq 0x458(%rax)
-0352 [ 4] mov 0x18(%rsp),%ecx
-0356 [ 2] inc %ebp
-0358 [ 4] add $0xc,%rbx
-035c [ 4] add %ecx,0xc(%rsp)
-0360 [ 3] cmp %r12d,%ebp
-0363 [ 6] jl <+0x90>
-0369 [ 5] jmp <+0x14f>
-036e [ 2] xchg %ax,%ax
+01f4 [ 5] jmp <+0xc0>
+01f9 [ 7] nopl 0x0(%rax)
+0200 [ 5] lea 0x20(%rsp),%r15
+0205 [ 9] movq $0x0,0x20(%rsp)
+020e [ 2] xchg %ax,%ax
+0210 [ 7] mov 0xbc8(%r14),%edi
+0217 [ 5] mov $0x8,%edx
+021c [ 3] mov %r15,%rsi
+021f [ 5] call <read@plt>
+0224 [ 4] cmp $0x8,%rax
+0228 [ 2] je <+0x210>
+022a [ 7] mov 0x190(%r14),%rax
+0231 [ 4] incl 0xc(%rsp)
+0235 [ 7] incq 0x440(%rax)
+023c [ 5] jmp <+0xc0>
+0241 [11] data16 cs nopw 0x0(%rax,%rax,1)
+024c [ 4] nopl 0x0(%rax)
+0250 [ 7] mov 0x188(%r14),%rax
+0257 [ 7] movzbl 0x11460(%rax),%eax
+025e [ 2] test %al,%al
+0260 [ 6] jne <+0xc0>
+0266 [ 7] mov 0xaa0(%r14),%r15d
+026d [ 3] test %r15d,%r15d
+0270 [ 6] js <+0xc0>
+0276 [ 8] movl $0x0,0x18(%rsp)
+027e [ 2] xor %edx,%edx
+0280 [ 2] xor %esi,%esi
+0282 [ 5] mov $0x80800,%ecx
+0287 [ 3] mov %r15d,%edi
+028a [ 5] call <accept4@plt>
+028f [ 2] test %eax,%eax
+0291 [ 6] jns <+0x370>
+0297 [ 5] call <__errno_location@plt>
+029c [ 2] mov (%rax),%eax
+029e [ 3] cmp $0x4,%eax
+02a1 [ 6] jne <+0x32f>
+02a7 [ 2] jmp <+0x27e>
+02a9 [ 7] nopl 0x0(%rax)
+02b0 [ 7] mov 0x190(%r14),%rax
+02b7 [ 5] mov 0x10(%rsp),%rdx
+02bc [ 7] lea <rip>(%rip),%rdi # <tomo_db0::Server::databases_>
+02c3 [ 2] mov (%rax),%esi
+02c5 [ 5] call <tomo_db0::DatabaseMap::consume_wake(unsigned int, tomo_db0::Ring&)>
+02ca [ 4] incl 0xc(%rsp)
+02ce [ 5] jmp <+0xc0>
+02d3 [11] data16 cs nopw 0x0(%rax,%rax,1)
+02de [ 2] xchg %ax,%ax
+02e0 [ 7] mov 0x188(%r14),%rax
+02e7 [ 7] movzbl 0x11460(%rax),%eax
+02ee [ 2] test %al,%al
+02f0 [ 6] jne <+0xc0>
+02f6 [ 7] mov 0xaa4(%r14),%r15d
+02fd [ 3] test %r15d,%r15d
+0300 [ 6] js <+0xc0>
+0306 [ 8] movl $0x0,0x18(%rsp)
+030e [ 2] xor %edx,%edx
+0310 [ 2] xor %esi,%esi
+0312 [ 5] mov $0x80800,%ecx
+0317 [ 3] mov %r15d,%edi
+031a [ 5] call <accept4@plt>
+031f [ 2] test %eax,%eax
+0321 [ 2] jns <+0x350>
+0323 [ 5] call <__errno_location@plt>
+0328 [ 2] mov (%rax),%eax
+032a [ 3] cmp $0x4,%eax
+032d [ 2] je <+0x30e>
+032f [ 3] cmp $0xb,%eax
+0332 [ 2] je <+0x342>
+0334 [ 7] mov 0x190(%r14),%rax
+033b [ 7] incq 0x458(%rax)
+0342 [ 4] mov 0x18(%rsp),%ecx
+0346 [ 4] add %ecx,0xc(%rsp)
+034a [ 5] jmp <+0xc0>
+034f [ 1] nop
+0350 [ 5] mov $0x9,%edx
+0355 [ 2] mov %eax,%esi
+0357 [ 3] mov %r14,%rdi
+035a [ 4] incl 0x18(%rsp)
+035e [ 5] call <void tomo_db0::IoLoop::admit_fd<true, false>(int, tomo_db0::UrKind)>
+0363 [ 2] jmp <+0x30e>
+0365 [11] data16 cs nopw 0x0(%rax,%rax,1)
 0370 [ 5] mov $0x1,%edx
 0375 [ 2] mov %eax,%esi
-0377 [ 3] mov %r15,%rdi
+0377 [ 3] mov %r14,%rdi
 037a [ 4] incl 0x18(%rsp)
 037e [ 5] call <void tomo_db0::IoLoop::admit_fd<true, false>(int, tomo_db0::UrKind)>
-0383 [ 2] jmp <+0x31e>
-0385 [11] data16 cs nopw 0x0(%rax,%rax,1)
-0390 [ 5] mov $0x9,%edx
-0395 [ 2] mov %eax,%esi
-0397 [ 3] mov %r15,%rdi
-039a [ 4] incl 0x18(%rsp)
-039e [ 5] call <void tomo_db0::IoLoop::admit_fd<true, false>(int, tomo_db0::UrKind)>
-03a3 [ 5] jmp <+0x23e>
-03a8 [ 5] call <__stack_chk_fail@plt>
+0383 [ 5] jmp <+0x27e>
+0388 [ 5] call <__stack_chk_fail@plt>
